src/shared_utilities/general_utils.jl

```diff
@@ -0,0 +1,18 @@
+export searchsortednearest
+"""
+    searchsortednearest(a, x)
+
+Find the index corresponding to the nearest value to `x` in `a`.
+"""
+function searchsortednearest(a, x)
```

Comment:
are we assuming that the data timeseries will be in order? That is a great assumption and requirement! not sure if it is obvious enough that we need to document it somewhere in a docstring?

Reply:
Added